gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -83,31 +114,70 @@
   @Override
   public List<WorkUnit> getWorkunits(SourceState state) {
     Map<String, List<WorkUnit>> workUnits = Maps.newHashMap();
-    Closer closer = Closer.create();
-    try {
-      KafkaWrapper kafkaWrapper = closer.register(KafkaWrapper.create(state));
-      List<KafkaTopic> topics = getFilteredTopics(kafkaWrapper, state);
-      for (KafkaTopic topic : topics) {
-        workUnits.put(topic.getName(), getWorkUnitsForTopic(kafkaWrapper, topic, state));
-      }
 
-      // Create empty WorkUnits for skipped partitions (i.e., partitions that have previous offsets,
-      // but aren't processed.
-      createEmptyWorkUnitsForSkippedPartitions(workUnits, state);
+    this.kafkaWrapper = KafkaWrapper.create(state);
```

Comment:
Since I made `kafkaWrapper` an instance variable I have to assign it to `this.kafkaWrapper` first, and then register it. I don't think I can assign the variable and register it at the same time?